Editorial review: Document interest invokers (#42021)
mdn/content: files/en-us/web/css/reference/properties/interest-delay-end/index.md

---

{{SeeCompatTable}}

The **`interest-delay-end`** [CSS](/en-US/docs/Web/CSS) property specifies a delay between the user losing interest in an [interest invoker](/en-US/docs/Web/API/Popover_API/Using_interest_invokers) element and the {{domxref("HTMLElement.loseinterest_event", "loseinterest")}} event being fired, which typically ends an effect on the target.

The `interest-delay-end` and {{cssxref("interest-delay-start")}} properties can both be set using the {{cssxref("interest-delay")}} shorthand.

## Syntax

```css
/* Single value */
interest-delay-end: normal;
interest-delay-end: 2s;
interest-delay-end: 250ms;

/* Global values */
interest-delay-end: inherit;
interest-delay-end: initial;
interest-delay-end: revert;
interest-delay-end: revert-layer;
interest-delay-end: unset;
```

### Values

The `interest-delay-end` property value is specified using the keyword `normal` or a {{cssxref("&lt;time>")}} value:

- `normal`
  - : The initial value; specifies that the delay should be set to the browser's default value.
- {{cssxref("&lt;time>")}}
  - : Sets the delay to a specific duration.

## Formal definition

{{CSSInfo}}

## Formal syntax

{{csssyntax}}

## Examples

> ### Basic `interest-delay-end` effect

Reviewer: ```suggestion
### Creating a basic `interest-delay-end` effect
```

In this example, we demonstrate how `interest-delay-end` affects interest invoker behavior.

#### HTML

We include a {{htmlelement("button")}}, a {{htmlelement("p")}}, and an {{htmlelement("input")}} of type `checkbox`. We specify the `<button>` as an interest invoker by giving it an `interestfor` attribute; its value is given as the `id` of the `<p>` element, therefore the paragraph is the invoker target. The paragraph is made into a popover by giving it a [`popover`](/en-US/docs/Web/HTML/Reference/Global_attributes/popover) attribute, causing it to be initially hidden.

```html live-sample___interest-invoker-delay
<button interestfor="mypopover">a button</button>
<p id="mypopover" popover>A hover toolip</p>
<form>
  <label for="apply-delay">Apply <code>interest-delay-end: 2s</code></label>
  <input type="checkbox" id="apply-delay" />
</form>

> ```

Reviewer: ```suggestion
```html live-sample___interest-invoker-delay
<button interestfor="mypopover">Button</button>
<p id="mypopover" popover>Hover tooltip</p>
<form>
  <input type="checkbox" id="apply-delay" />
  <label for="apply-delay">Apply an <code>interest-delay-end</code> of <code>2s</code></label>
</form>
```
```

#### CSS

> In the CSS, we specify a rule with a `.delay` selector, which will apply an `interest-delay-end` value of `2s` to any interest invoker the `delay` class is set on. We will set this on the `<button>` when the checkbox is checked using JavaScript.

Reviewer: ```suggestion
In the CSS, we specify a rule with a `.delay` selector that applies an `interest-delay-end` value of `2s` to any interest invoker the `delay` class is set on. We will set this on the `<button>` when the checkbox is checked using JavaScript.
```
